### ISSUE TITLE
Support sizes=auto

### ISSUE THREAD

**Issue by eeeps** [ausi/respimagelint#93]
Currently, when running the tool on a page that uses the [brand-new `sizes=auto` syntax](https://chromestatus.com/feature/5191555708616704), RespImageLint complains,

> The `sizes` attribute has to match the width of the image
> The size of the image doesn’t match the `sizes` attribute `auto`

At a minimum, `auto` should be an allowed `sizes` value. A better solution would check that images with `sizes=auto` were also lazy-loaded, [and had an extrinsic width + height](https://ericportis.com/posts/2023/auto-sizes-pretty-much-requires-width-and-height/), showing informative errors/warnings if not.

**Comment by TechAurelian2** [comment]
Maybe this issue should also be opened in the repository of the alternative Chrome extension recommended in the README of this project:

<https://github.com/peter-neumann-dev/responsive-image-linter>

**Comment by ausi** [comment]
@eeeps so `sizes=auto` is only allowed together with `loading=lazy`, right?

Regarding extrinsic width and height, do you think the linter should report this as an error if they are missing?

**Comment by ausi** [comment]
`sizes=auto` is not yet shipped by any browser, right?

**Comment by peter-neumann-dev** [comment]
> Maybe this issue should also be opened in the repository of the alternative Chrome extension recommended in the README of this project:
> 
> https://github.com/peter-neumann-dev/responsive-image-linter

@TechAurelian2 
I think we should not duplicate the issues in general. The Chrome extension is just a wrapper around the linter. 

Issues regarding the extension itself are of course part of the extension's repo (that's how the issue template is asking for as well). I would add a hint that issues related to the linter are listed in this repository 🙂

**Comment by eeeps** [comment]
@ausi [Auto-sizes is shipping in Chrome 126](https://wpt.fyi/results/html/semantics/embedded-content/the-img-element/sizes?label=master&product=chrome%5Bexperimental%5D&product=chrome%5Bstable%5D&product=firefox%5Bexperimental%5D&product=safari%5Bexperimental%5D&product=chrome%5Bbeta%5D&aligned&q=sizes%20auto) which [should go stable... tomorrow!](https://chromiumdash.appspot.com/schedule) (Tuesday)

[`loading=lazy` is required for auto-sizes](https://html.spec.whatwg.org/multipage/images.html#valdef-sizes-auto). Without it, `auto` is ignored.

Note that `auto` can be a part of a `sizes` that includes multiple values (for fallback purposes in browsers that don't support auto), but it can only be the first value. So `sizes="auto, 80vw"` is valid but `sizes="(min-width: 800px) 800px, auto"` is not.

I think the linter should report an error if there is no extrinsic height + extrinsic width, height+aspect-ratio, or (most commonly) width+aspect-ratio. Without these, the image will fall back to an [imposed 2:1 intrinsic aspect ratio](https://html.spec.whatwg.org/multipage/rendering.html#img-contain-size), which probably isn't what people want.

`<img height width>` in HTML + `width: 100%; height: auto;` in CSS is probably the easiest way to do this, but is not the only way; the linter should allow for other intentional cases like:

```
.hero-image {
  width: 100%;
  aspect-ratio: 4 / 3;
}

<img loading=lazy class=hero-img sizes=auto srcset="big 1000w, little 500w">
```

I haven't given a lot of thought as to how this could or should be checked.

**Comment by ausi** [comment]
> At a minimum, `auto` should be an allowed `sizes` value. A better solution would check that images with `sizes=auto` were also lazy-loaded,…

Done in cb36c068c7244df695cae30892e6a533c2082947
It checks now for `loading="lazy"` and ensures the attribute starts with `auto` or `auto,`.

> I think the linter should report an error if there is no extrinsic height + extrinsic width, height+aspect-ratio, or (most commonly) width+aspect-ratio.

This is still missing, therefore keeping this issue open.

**Comment by peter-neumann-dev** [comment]
I will try to update the Chrome extension within the next days as well then. 🙂

Thanks, @ausi

**Comment by ausi** [comment]
See also https://mastodon.social/@ausi/115020191123515279 the linter needs to be made aware that `sizes` is optional on `<source>` if `<img sizes=auto>` (but IMO only after `sizes=auto` becomes a baseline)
